network/nxos/nxos_nxapi.py

```diff
@@ -31,238 +31,259 @@
-    https_port:
-        description:
-            - Port on which the HTTPS server will listen.
-        required: false
-        default: 443
-    http:
-        description:
-            - Enable/disable HTTP server.
-        required: false
-        default: true
-        choices:
-            - true
-            - false
-        aliases:
-            - enable_http
-    https:
-        description:
-            - Enable/disable HTTPS server.
-        required: false
-        choices:
-            - true
-            - false
-        default: true
-        aliases:
-            - enable_https
-    sandbox:
-        description:
-            - Enable/disable NXAPI web based UI for entering commands.
-        required: false
-        default: true
-        choices:
-            - true
-            - false
-        aliases:
-            - enable_sandbox
+  state:
+    description:
+      - Set to started or stopped. A state of started will
+        enable NXAPI access, and a state of stopped will
+        disable or shutdown all NXAPI access.
+    choices:
+      - present
+      - absent
+    required: false
+    default: present
+  http_port:
+    description:
+      - Port on which the HTTP server will listen.
+    required: false
+    default: 80
+  https_port:
+    description:
+      - Port on which the HTTPS server will listen.
+    required: false
+    default: 443
+  http:
+    description:
+      - Enable/disable HTTP server.
+    required: false
+    default: true
+    choices:
+      - true
+      - false
+    aliases:
+      - enable_http
+  https:
+    description:
+      - Enable/disable HTTPS server.
+    required: false
+    choices:
+      - true
+      - false
+    default: true
+    aliases:
+      - enable_https
+  sandbox:
+    description:
+      - Enable/disable NXAPI web based UI for entering commands.
+    required: false
+    default: true
+    choices:
+      - true
+      - false
+    aliases:
+      - enable_sandbox
+  config:
+    description:
+      - Specifies the device running-config to be used
+    required: false
+    default: null
+    version_added: "2.2"
 """
 
 EXAMPLES = """
-    - name: Enable NXAPI access with default configuration
-      nxos_nxapi:
-          provider: {{ provider }}
-
-    - name: Enable NXAPI with no HTTP, HTTPS at port 9443 and sandbox disabled
-      nxos_nxapi:
-          enable_http: false
-          https_port: 9443
-          enable_sandbox: no
-          provider: {{ provider }}
-
-    - name: shutdown NXAPI access
-      nxos_nxapi:
-          state: stopped
-          provider: {{ provider }}
+# Note: examples below use the following provider dict to handle
+#       transport and authentication to the node.
+vars:
+  cli:
+    host: "{{ inventory_hostname }}"
+    username: admin
+    password: admin
+
+- name: Enable NXAPI access with default configuration
+  nxos_nxapi:
+    provider: {{ cli }}
+
+- name: Enable NXAPI with no HTTP, HTTPS at port 9443 and sandbox disabled
```

Comment:
From reading `set_https` it and like you do need to set 'enable_https: true'. Setting a port doesn't audio enable it. 

Http is `default=True`
Https is `default=False`